### PR TITLE
Raise unused kwargs image processor

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

This PR captures all kwargs passed to an `ImageProcessor` `preprocess` method and compares them to what's expected, raising an exception or logging an informative message when a difference is found.
This will 
1) Make preprocess methods more reliable
2) Inform users when an expected kwargs has no actual impact in the preprocessing

- [x] Did you write any new necessary tests?

## Who can review?

- vision models: @amyeroberts

### CODE REVIEW

**Comment by molbap** [comment]
Finishing up draft, removing this validation functionality from #28711

**Comment by molbap** [comment]
Main concern I had was using `inspect`, so I moved it to the test suite instead of the actual validation function. That way we don't inspect at runtime, we just compare static lists. For new model additions, `self._valid_processor_keys` remain optional (if it's not there, nothing is run) but it'd be a recommendation.
As to what the validation raises, for now it's a `logger.info`, could be a warning instead but should not raise an exception as a good chunk of existing models /processors from the hub would break down

File: src/transformers/models/fuyu/image_processing_fuyu.py [hunk 1]
```diff
@@ -260,6 +260,24 @@ def __init__(
         self.do_rescale = do_rescale
         self.rescale_factor = rescale_factor
         self.patch_size = patch_size if patch_size is not None else {"height": 30, "width": 30}
+        self._valid_processor_keys = [
```
Comment (amyeroberts) [inline src/transformers/models/fuyu/image_processing_fuyu.py:263]: Missing validation in the preprocess call for this model?

File: src/transformers/image_utils.py [hunk 2]
```diff
@@ -718,3 +718,11 @@ def validate_annotations(
...
                 "the latter being a list of annotations in the COCO format."
             )
+
+
+def validate_kwargs(valid_processor_keys: List[str], captured_kwargs: List[str]):
+    unused_keys = set(captured_kwargs).difference(set(valid_processor_keys))
+    if unused_keys:
+        unused_key_str = ", ".join(unused_keys)
+        # TODO raise a warning here instead of simply logging?
+        logger.info(f"Unused or unrecognized kwargs: {unused_key_str}.")
```
Comment (amyeroberts) [inline src/transformers/image_utils.py:728]: Yep, we can set `logger.warning` to encourage users to pass in the correct kwargs here

**Review by amyeroberts — APPROVED** [review]
Looks great - thanks for adding this! ❤️

File: src/transformers/models/fuyu/image_processing_fuyu.py (hunk 1 above)
Comment (molbap) [reply src/transformers/models/fuyu/image_processing_fuyu.py:263]: ah, yes, well there is no `**kwargs` in this one so no need for validation I suppose?

Comment (amyeroberts) [reply src/transformers/models/fuyu/image_processing_fuyu.py:263]: Ah, yes, in that case there's no need :)

**Pull request merged**